ui/packages/atlasmap/src/UI/AddFieldTypeahead.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { FunctionComponent, ChangeEvent, useCallback } from "react";
+import {
+  Select,
+  SelectOption,
+  Text,
+  TextContent,
+  TextVariants,
+  SelectProps,
+} from "@patternfly/react-core";
+import { useToggle } from "./useToggle";
+
+export interface IAddFieldTypeaheadField {
+  label: string;
+  info?: string;
+  onAdd: () => void;
+}
+
+export interface IAddFieldTypeaheadProps {
+  fields: IAddFieldTypeaheadField[];
+  ariaLabelTypeAhead: string;
+  placeholderText: string;
+}
+
+export const AddFieldTypeahead: FunctionComponent<IAddFieldTypeaheadProps> = ({
+  fields,
+  ariaLabelTypeAhead,
+  placeholderText,
+  ...props
+}) => {
+  const { state, toggle, toggleOff } = useToggle(false);
+
+  const renderOptions = (fields: IAddFieldTypeaheadField[]) => {
+    return fields.map((f, index) => (
+      <SelectOption
```

Comment:
The select options for fields should list the document name (unselectable) first followed by selectable fields in that document as opposed listing the document for each field redundantly.

Reply:
It does
![image](https://user-images.githubusercontent.com/966316/81946698-f94d1900-95ff-11ea-853d-4204934e0630.png)

Reply:
Only one JSONInstanceSource - like this:
![sc1](https://user-images.githubusercontent.com/2981421/81950418-e58abe00-95d1-11ea-947f-c255d261ddf6.png)

Reply:
That's not how grouped selects should be done with PF4. Check the example here https://www.patternfly.org/v4/documentation/react/components/select#grouped-single

Reply:
It's close.  Anyway - this can be tackled as a PF4 grouped-single as another issue.

Reply:
There now, PF4 style:
![image](https://user-images.githubusercontent.com/966316/81956458-9eb9ba00-960b-11ea-950c-bf3fd1de0ded.png)